crates/server/src/store/recursor/authority.rs

```diff
@@ -172,4 +172,12 @@ impl LookupObject for RecursiveLookup {
     fn take_additionals(&mut self) -> Option<Box<dyn LookupObject>> {
         None
     }
+
+    fn dnssec_validated(&self) -> bool {
+        // XXX what should be the behavior if the lookup yields no records?
```

Comment:
Does this need discussion?

Reply:
I think it needs a better understanding of the RFCs and an accompanying conformance test.

I think the condition mentioned here would happen if a query yields an empty answer section (NXDOMAIN), DNSSEC validation is enabled but both NSEC and NSEC3 are disabled. "does the AD flag get set in the response or not?" is what needs to be researched, either in RFCs or by testing what other implementations do.

I have that in my TODO list. I can also open issues for missing / pending conformance tests if you think that's worthwhile.

Reply:
If it's on your list I'm fine merging it as is. Just wanted to mark it with a discussion to get a bit more context on how/when this will be followed up on.